source/Calamari/Commands/KubernetesAuthenticationCommand.cs

```diff
@@ -0,0 +1,70 @@
+#if !NET40
+using System.Collections.Generic;
+using Calamari.Common.Commands;
+using Calamari.Common.Features.Processes;
+using Calamari.Common.Plumbing.FileSystem;
+using Calamari.Common.Plumbing.Logging;
+using Calamari.Common.Plumbing.Variables;
+using Calamari.Kubernetes;
+using Calamari.Kubernetes.Integration;
+
+namespace Calamari.Commands
+{
+    [Command("authenticate-to-kubernetes")]
+    public class KubernetesAuthenticationCommand: Command<KubernetesAuthenticationCommandInput>
+    {
+        private readonly ILog log;
+        private readonly IVariables variables;
+        private readonly ICalamariFileSystem fileSystem;
+        private readonly ICommandLineRunner commandLineRunner;
+        private readonly Kubectl kubectl;
+
+        public KubernetesAuthenticationCommand(
+            ILog log,
+            IVariables variables,
+            ICalamariFileSystem fileSystem,
+            ICommandLineRunner commandLineRunner,
+            Kubectl kubectl)
+        {
+            this.log = log;
+            this.variables = variables;
+            this.fileSystem = fileSystem;
+            this.commandLineRunner = commandLineRunner;
+            this.kubectl = kubectl;
+        }
+
+        protected override void Execute(KubernetesAuthenticationCommandInput inputs)
+        {
+            log.Info("Setting up KubeConfig authentication.");
+            var environmentVars = new Dictionary<string, string>();
+            var runningDeployment = new RunningDeployment(variables);
+            
+            kubectl.SetEnvironmentVariables(environmentVars);
+            kubectl.SetWorkingDirectory(runningDeployment.CurrentDirectory);
+            
+            var setupKubectlAuthentication = new SetupKubectlAuthentication(variables,
+                log,
+                commandLineRunner,
+                kubectl,
+                environmentVars,
+                runningDeployment.CurrentDirectory);
+            var accountType = variables.Get("Octopus.Account.AccountType");
+
+            try
+            {
+                var result = setupKubectlAuthentication.Execute(accountType);
+                result.VerifySuccess();
+            }
+            catch (CommandLineException ex)
+            {
+                log.Error("KubeConfig authentication failed.");
+                throw new CommandException(ex.Message);
+            }
+            
+            variables.Set("Octopus.KubeConfig.Path", fileSystem.GetFullPath(environmentVars["KUBECONFIG"]));
```

Comment:
Basically the commands communicate among each other using special variables, like this one `Octopus.KubeConfig.Path`. One command sets this and a following command reads this. I think using a string literal here will be harder to maintain after we finish our work.

In this PR it is still easy to figure out which command sets this variable and which reads this variable. But I imagine once we merge this PR, we'll lose track. Let's extract these special variable name strings to the `SpecialVariables` class. In this way, if another developers want to figure out where a certain variables was set, he can do so by "Find usages" on that special variable, which is easier.